### PR TITLE
Enable symfony 4

### PR DESCRIPTION
**Pull request by caciobanu**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,17 +22,18 @@
...
         "simple-bus/jms-serializer-bundle-bridge": "~3.0",
-        "symfony/console": "~2.0 || ~3.0",
-        "symfony/finder": "~2.0 || ~3.0",
-        "symfony/process": "~2.0 || ~3.0",
-        "symfony/translation": "~2.0 || ~3.0"
+        "symfony/console": "~2.7 || ~3.3 || ~4.0",
+        "symfony/finder": "~2.7 || ~3.3 || ~4.0",
+        "symfony/process": "~2.7 || ~3.3 || ~4.0",
+        "symfony/translation": "~2.7 || ~3.3 || ~4.0",
+        "symfony/stopwatch": "~2.7 || ~3.3 || ~4.0"
```
Comment (ruudk) [inline composer.json]: Asked JMS for a new tag, so we can remove the stopwatch requirement. https://github.com/schmittjoh/JMSSerializerBundle/issues/627

Comment (caciobanu) [reply composer.json]: I'll remove it when a new tag is made.

**Comment by caciobanu** [comment]
Build failure seems related to https://github.com/php-amqplib/RabbitMqBundle/pull/500

**Comment by caciobanu** [comment]
https://github.com/php-amqplib/RabbitMqBundle/pull/500 merged. Waiting for issue https://github.com/php-amqplib/RabbitMqBundle/issues/503 to be resolved and I'll have another try on this PR.

**Comment by caciobanu** [comment]
https://github.com/php-amqplib/RabbitMqBundle/issues/503 closed. I will update this PR when I'll have the necessary time.

File: composer.json (hunk 1 above)
Comment (ruudk) [reply composer.json]: It's tagged!

**Comment by caciobanu** [comment]
Build still failing due to https://github.com/php-amqplib/RabbitMqBundle/pull/509 &#128546;.

**Comment by caciobanu** [comment]
@ruudk could we get this merged and get a new tag ?

This is not anymore under our control and we shouldn't wait for https://github.com/php-amqplib/RabbitMqBundle/pull/509 as it impacts only the tests.

**Comment by cmodijk** [comment]
@caciobanu Can you explain the non descriptive error i see is `The log file does not contain "Notified of message"`. I can't tell what the command error is from looking at the CI server.

**Comment by caciobanu** [comment]
When a command or an event is consumed it should log a message to a file: https://github.com/SimpleBus/RabbitMQBundleBridge/blob/master/tests/Functional/SimpleBusRabbitMQBundleTest.php#L72. 

The problem is that the command run in background on Travis to setup the necessary queues and exchanges fails on Symfony 3.4 and 4.x because a service used in that command is private by the default; this why I've made https://github.com/php-amqplib/RabbitMqBundle/pull/509. 

This command is run before actually publishing and consuming messages: https://github.com/SimpleBus/RabbitMQBundleBridge/blob/master/tests/Functional/SimpleBusRabbitMQBundleTest.php#L42.

**Comment by cmodijk** [comment]
@caciobanu Thanks for the explanation. The PHP 7.1 with `no environment variables set` is also using Symfony 4?

**Comment by cmodijk** [comment]
So without https://github.com/php-amqplib/RabbitMqBundle/pull/509 only the tests fail? If you install it without the `RabbitMqBundle` fix inside a APP it works?

**Comment by caciobanu** [comment]
Yes, PHP 7.1 without env vars uses Symfony 4.latest on Travis build.

Yes, only the tests fail. It should work. The fix is for a Symfony command inside the RabbitMqBundle which does not affect this bridge.

**Comment by cmodijk** [comment]
Is it related to the `setup-fabric` command?

**Comment by caciobanu** [comment]
Yes.

**Comment by cmodijk** [comment]
Hmm, oke that would mean that we support Symfony 4 but the moment you try to setup your RabbitMQ configuration this error would happen. In our case we run the `setup-fabric` during every release to setup al our settings when changed happen. Should we then also not wait until they fix the problem and version constraint it in `composer.json` ? 

I'm split at this point i think you are right in then we should merge and tag because its not our problem but stil it feels a bit broken.

**Comment by caciobanu** [comment]
From my point of view that problem is in the RabbitMqBundle. That command is not used in this bridge, except for tests, and so it should not be handled here.

**Comment by cmodijk** [comment]
Merged / Tagged https://github.com/SimpleBus/RabbitMQBundleBridge/releases/tag/v4.0.1

**Comment by caciobanu** [comment]
Thank you guys!

**Comment by cmodijk** [comment]
Thank you for taking the time to make the pull request 👍

**Pull request merged**